crates/sgx_validation/src/lib.rs

```diff
@@ -75,42 +83,68 @@ impl Sgx {
         //
         // Additionally, we do this check early to be defensive.
         if cri.public_key.algorithm != pck.subject_public_key_info.algorithm {
-            return Err(anyhow!("sgx pck algorithm mismatch"));
+            bail!("sgx pck algorithm mismatch");
         }
 
         if !dbg {
             // TODO: Validate that the certification request came from an SGX enclave.
-            let hash = Sha256::digest(&cri.public_key.to_vec()?);
-            if hash.as_slice() != &rpt.reportdata[..hash.as_slice().len()] {
-                return Err(anyhow!("sgx report data is invalid"));
-            }
-
-            if rpt.mrenclave != [0u8; 32] {
-                return Err(anyhow!("untrusted enarx runtime"));
-            }
-
-            if rpt.mrsigner != [0u8; 32] {
-                return Err(anyhow!("untrusted enarx signer"));
-            }
-
-            if rpt.cpusvn != [0u8; 16] {
-                return Err(anyhow!("untrusted cpu"));
-            }
-
-            if rpt.attributes() != Attributes::default() {
-                return Err(anyhow!("untrusted attributes"));
+            let hash = Sha256::digest(cri.public_key.to_vec()?);
+            ensure!(
+                hash.as_slice() == &rpt.reportdata[..hash.as_slice().len()],
+                "sgx report data is invalid"
+            );
+
+            if let Some(config) = config {
+                let config = &config.config;
+                if !config.enarx_signer.is_empty() {
+                    let mut signed = false;
+                    for signer in &config.enarx_signer {
+                        if rpt.mrsigner == signer.as_ref() {
+                            signed = true;
+                            break;
+                        }
+                    }
+                    ensure!(signed, "sgx untrusted enarx signer");
+                }
+
+                if !config.enarx_hash.is_empty() {
+                    let mut signed = false;
+                    for hash in &config.enarx_hash {
+                        if rpt.mrenclave == hash.as_ref() {
+                            signed = true;
+                            break;
+                        }
+                    }
+                    ensure!(signed, "sgx untrusted enarx hash");
+                }
+
+                if let Some(product_id) = config.enclave_product_id {
+                    ensure!(
+                        rpt.enclave_product_id() >= product_id,
+                        "sgx untrusted enclave product id"
+                    );
+                }
+
+                if let Some(version) = config.enclave_security_version {
+                    ensure!(
+                        rpt.enclave_security_version() >= version,
+                        "sgx untrusted enclave security version"
+                    );
+                }
+
+                if config.features > 0
```

Comment:
```suggestion
                if config.features != 0
```
would be clearer, since that's not a decimal

Reply:
Fixed